static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,86 @@
+import {useCallback, useContext, useEffect} from 'react';
+import styled from '@emotion/styled';
+
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import {DrawerContext} from 'sentry/components/globalDrawer/context';
+import {useLocation} from 'sentry/utils/useLocation';
+
+export interface DrawerOptions {
+  //
+  // If true (default), closes the drawer when an escape key is pressed
+  //
+  closeOnEscapeKeypress?: boolean;
+  //
+  // If true (default), closes the drawer when anywhere else is clicked
+  //
+  closeOnOutsideClick?: boolean;
+  //
+  // Callback for when the drawer closes
+  //
+  onClose?: () => void;
+  //
+  // Callback for when the drawer opens
+  //
+  onOpen?: () => void;
+}
+
+export interface DrawerRenderProps {
+  //
+  // Body container for the drawer
+  //
+  Body: typeof DrawerBody;
+  //
+  // Close the drawer
+  //
+  closeDrawer: () => void;
+}
+
+interface GlobalDrawerProps {}
+
+export default function GlobalDrawer(_props: GlobalDrawerProps) {
```

Comment:
I think you can collapse the context provider stuff just into this component

Reply:
Like, the only component using this component `GlobalDrawer` is `DrawerContextProvider`. So why have this be separate and read the context, when all this components code and just be inside of `DrawerContextProvider`

Reply:
That makes sense, I've refactored it into one component, thanks!